docs/dev/README.md

```diff
@@ -0,0 +1,102 @@
+```yaml
+OS_IMAGES: '[{"openshift_version":"4.16","cpu_architecture":"x86_64","url":"https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.16/4.16.3/rhcos-4.16.3-x86_64-live.x86_64.iso","version":"416.94.202406251923-0"}]'
+RELEASE_IMAGES: '[{"openshift_version":"4.16","cpu_architecture":"x86_64","cpu_architectures":["x86_64"],"url":"quay.io/openshift-release-dev/ocp-release:4.16.10-x86_64","version":"4.16.10","default":true}]' 
+```
+
+We can start this local environment using make:
+```shell
+make deploy-onprem
+```
+Once finished, we can see the containers running:
+```shell
+$ podman ps
+
+CONTAINER ID  IMAGE                                                      COMMAND               CREATED        STATUS        PORTS                                                                   NAMES
+487e6c1bdb9a  localhost/podman-pause:4.9.3-1708357294                                          4 minutes ago  Up 3 minutes  0.0.0.0:8080->8080/tcp, 0.0.0.0:8090->8090/tcp, 0.0.0.0:8888->8888/tcp  7c18ebd0915a-infra
+8479b5eb8a8d  quay.io/sclorg/postgresql-12-c8s:latest                    run-postgresql        4 minutes ago  Up 3 minutes  0.0.0.0:8080->8080/tcp, 0.0.0.0:8090->8090/tcp, 0.0.0.0:8888->8888/tcp  assisted-installer-db
+ffb9013c4fab  quay.io/edge-infrastructure/assisted-installer-ui:latest   /deploy/start.sh      3 minutes ago  Up 3 minutes  0.0.0.0:8080->8080/tcp, 0.0.0.0:8090->8090/tcp, 0.0.0.0:8888->8888/tcp  assisted-installer-ui
+100c865abfd6  quay.io/edge-infrastructure/assisted-image-service:latest  /assisted-image-s...  3 minutes ago  Up 3 minutes  0.0.0.0:8080->8080/tcp, 0.0.0.0:8090->8090/tcp, 0.0.0.0:8888->8888/tcp  assisted-installer-image-service
+78924b68f7af  quay.io/edge-infrastructure/assisted-service:latest        /assisted-service     3 minutes ago  Up 3 minutes  0.0.0.0:8080->8080/tcp, 0.0.0.0:8090->8090/tcp, 0.0.0.0:8888->8888/tcp  assisted-installer-service
+```
+For testing we can use `curl`:
+```shell
+$ curl -i http://127.0.0.1:8090/api/assisted-install/v2/infra-envs
+HTTP/1.1 200 OK
+Content-Type: application/json
+Vary: Accept-Encoding
+Date: Wed, 11 Sep 2024 14:16:43 GMT
+Content-Length: 3
+
+[]
+
+$ curl -i http://127.0.0.1:8090/api/assisted-install/v2/clusters
+HTTP/1.1 200 OK
+Content-Type: application/json
+Vary: Accept-Encoding
+Date: Wed, 11 Sep 2024 14:18:59 GMT
+Content-Length: 3
+
+[]
+```
+A more detailed description of how to use the REST API is available in the [user guide](docs/user-guide/rest-api-getting-started.md).
+
+## assisted-test-infra
+[Detailed documentation](https://github.com/openshift/assisted-test-infra/blob/master/README.md)
+
+This scenario is useful for developing and testing our end-to-end tests.
+This environment requires a large amount of resources since it will deploy additional baremetal hosts (VMs), so **running it in local environments is not recommended, do it on a beaker box or an ephemeral server**.
```

Comment:
The main reason it's not recommended to run in local envs is not about resource usage, but it will change some host configuration that might be relevant for the user. Maybe worth mentioning? Maybe something like `Assisted Test infra is an opinionated framework and running it in local environments is not recommended as it will change settings to the host`